silkworm/core/chain/config.hpp

```diff
@@ -57,34 +58,124 @@ struct ChainConfig {
     //! \brief Returns the type of the (pre-Merge) protocol rule set
     protocol::RuleSetType protocol_rule_set{protocol::RuleSetType::kNoProof};
 
+    const std::optional<BlockNum>& homestead_block() const{
```

Comment:
Are those accessor really being used anywhere?

Reply:
Yes, there are some places in the code where those accessors are called individually and we are throwing an exception if the ruleset used is not kTrust